aptos-move/framework/aptos-framework/sources/multisig_account.move

```diff
@@ -319,32 +325,46 @@ module aptos_framework::multisig_account {
+        let (num_approvals, _) = num_approvals_and_rejections(multisig_account, sequence_number);
+        sequence_number == last_resolved_sequence_number(multisig_account) + 1 &&
+            num_approvals >= num_signatures_required(multisig_account)
+    }
+
+    #[view]
+    /// Return true if the owner can execute the transaction with given transaction id now.
+    public fun can_execute(owner: address, multisig_account: address, sequence_number: u64): bool acquires MultisigAccount {
+        assert_valid_sequence_number(multisig_account, sequence_number);
+        let (num_approvals, _) = num_approvals_and_rejections(multisig_account, sequence_number);
+        if (!has_voted_for_approval(multisig_account, sequence_number, owner)) {
+            num_approvals = num_approvals + 1;
+        };
+        is_owner(owner, multisig_account) &&
+            sequence_number == last_resolved_sequence_number(multisig_account) + 1 &&
+            num_approvals >= num_signatures_required(multisig_account)
     }
 
     #[view]
     /// Return true if the transaction with given transaction id can be officially rejected.
-    public fun can_be_rejected(
-        multisig_account: address, sequence_number: u64): bool acquires MultisigAccount {
-        let multisig_account_resource = borrow_global<MultisigAccount>(multisig_account);
-        assert!(
-            sequence_number > 0 && sequence_number < multisig_account_resource.next_sequence_number,
-            error::invalid_argument(EINVALID_SEQUENCE_NUMBER),
-        );
-        let transaction = table::borrow(&multisig_account_resource.transactions, sequence_number);
-        let (_, num_rejections) = num_approvals_and_rejections(&multisig_account_resource.owners, transaction);
-        sequence_number == multisig_account_resource.last_executed_sequence_number + 1 &&
-            num_rejections >= multisig_account_resource.num_signatures_required
+    public fun can_be_rejected(multisig_account: address, sequence_number: u64): bool acquires MultisigAccount {
+        assert_valid_sequence_number(multisig_account, sequence_number);
+        let (_, num_rejections) = num_approvals_and_rejections(multisig_account, sequence_number);
+        sequence_number == last_resolved_sequence_number(multisig_account) + 1 &&
+            num_rejections >= num_signatures_required(multisig_account)
+    }
+
+    #[view]
+    /// Return true if the owner can execute the "rejected" transaction with given transaction id now.
+    public fun can_reject(owner: address, multisig_account: address, sequence_number: u64): bool acquires MultisigAccount {
+        assert_valid_sequence_number(multisig_account, sequence_number);
+        let (_, num_rejections) = num_approvals_and_rejections(multisig_account, sequence_number);
+        if (!has_voted_for_rejection(multisig_account, sequence_number, owner)) {
+            num_rejections = num_rejections + 1;
+        };
```

Comment:
Isn't this saying if the owner has not voted for rejection, increment number of rejections?

Reply:
Right, "+1" is the implicit vote. The "execute" operation will involve an implicit vote for approval, and the "execute-reject" operation will involve an implicit vote for rejection.